File: Practice/CP31/1100/250 Thousand Tons of TNT.cpp
/*
	author: Ankesh
*/
#include<bits/stdc++.h>
using namespace std;
#define ll long long

void solve(){
	int n;
	cin>>n;

	vector<ll>a(n,0);
	for (ll it=0;it<n;it++){
		cin>>a[it];
	}
	ll ans = 0;
	for (int k = 1; k<=n; k++){

		if ( n % k != 0){
			continue;
		}

		int left = 0;
		int right = 0;
		ll sum = 0;
		ll mini = INT_MAX;
		ll maxi = INT_MIN;
		while (right < n){
			if ((right - left + 1) > k){
				left = right;
				mini = min(mini,sum);
				maxi = max(maxi,sum);
				sum = 0;
			}
			sum += a[right];
			right ++;
		}
		mini = min(mini,sum);
		maxi = max(maxi,sum);
		ans = max(ans,(ll)(maxi - mini));
	}
	cout<<ans<<endl;
}

int main(){

	ios_base::sync_with_stdio(false);
    cin.tie(NULL);

	#ifndef ONLINE_JUDGE
	freopen("input.txt","r",stdin);
	freopen("output.txt","w",stdout);
	#endif
	ll tc = 1;
	cin>>tc;
	while (tc-- ){
		solve();
	}
	return 0;
}

File: Practice/CP31/1100/250 Thousand Tons of TNT.py
import sys

# sys.stdin = open('input.txt','r')
# sys.stdout = open('output.txt','w')

def solve():
	n = int(input())
	a = list(map(int,input().split()))

	ans = 0
	for k in range(1,n+1):
		if (n % k) != 0:
			continue
		left = 0
		right = 0
		s = 0
		maxi = -float('inf')
		mini = float('inf')
		while right < n:
			if (right - left + 1) > k:
				maxi = max(maxi,s)
				mini = min(mini,s)
				s = 0
				left = right
			s += a[right]
			right += 1

		maxi = max(maxi,s)
		mini = min(mini,s)
		ans = max(ans,maxi - mini)
	print(ans)


tc = 1
tc = int(input())
while tc > 0:
	solve()
	tc -= 1
